### DISCUSSION TITLE
How to share conversation history to a tool in a SelectorGroupChat conversation

### DISCUSSION THREAD

**Discussion by 7899manjunath** [microsoft/autogen#4451] [Q&A]
At the end of the "SelectorGroupChat" conversation, we are planning to  do set of validations and store the summery of the conversation in json file. we are planning use an agent with a tool to achieve this. May I know how we can share entire conversation to tool?

**Comment by ekzhu** [comment]
I think you can use the `TaskResult` object returned by the run of `SelectorGroupChat` which contains the whole conversation history of the last run.

It may make sense to just use a single agent to perform the analysis of the result or just apply your application code directly without using agents, unless the post processing/ analytics can be abstracted as a multi-turn workflow.

**Comment by 7899manjunath** [comment]
@ekzhu  Scenario: We are starting a SelectorGroupChat with 'Developer', 'SeniorDeveloper', 'Architect' and 'Aggregator' agents.  developers and architect involve in group chat. Once they come to conclusion, we are planning select 'Aggregator' agent to summarize the conversation in the group chat. 'Aggregator' agent uses the tool to summarize the content. In this scenario, how tool will get the conversation history.

**Reply by ekzhu** [reply]
You can call the `on_messages` on the Aggregator agent with the complete conversation history from `TaskResult` filtering out tool call related messages (i.e., not `ChatMessage` type). 

What's is the purpose of the Aggregator agent, if all you want it to do is to call the tool on the conversation history? How about calling the tool directly with the conversation history from `TaskResult`?

**Reply by 7899manjunath** [reply]
Yes, want to call the tool on conversation history in 'Aggregator' agent. Aggregator is not only summarizing the conversation, going forward we are planning to use it for validating conversation, saving conversation in file and etc.

**Reply by ekzhu** [reply]
Maybe controversial but I generally recommend against using model-driven agents (e.g., `AssistantAgent`) if the steps of model client calls or tool calls are deterministic. You might want to consider custom agent that are rule-based.

For tasks like saving the conversation, you shouldn't be using agents at all.
